S3/Config.py

```diff
@@ -275,38 +281,46 @@ def role_refresh(self):
             except:
                 warning("Could not refresh role")
 
-    def env_config(self):
-        cred_content = ""
+    def aws_credential_file(self):
         try:
-            cred_file = open(os.environ['AWS_CREDENTIAL_FILE'],'r')
-            cred_content = cred_file.read()
+            aws_credential_file = os.path.expanduser('~/.aws/credentials') 
+            if 'AWS_CREDENTIAL_FILE' in os.environ and os.path.isfile(os.environ['AWS_CREDENTIAL_FILE']):
+                aws_credential_file = os.environ['AWS_CREDENTIAL_FILE']
+
+            config = PyConfigParser()
+
+            debug("Reading AWS credentials from", aws_credential_file)
+            config.read(aws_credential_file)
+
+            profile = "default"
+            if 'AWS_PROFILE' in os.environ:
+                profile = os.environ['AWS_PROFILE']
+            
+            debug("Using AWS profile '{}'".format(profile))
+
+            # trying to read aws_access_key_id from credentials file
+            profile_access_key = config.get(profile, 'aws_access_key_id')
+            debug('Setting "aws_access_key_id" from file {} as "access_key"'.format(aws_credential_file))
+            Config().update_option('access_key', config_unicodise(profile_access_key))
+
+            # trying to read aws_secret_access_key from credentials file
+            profile_secret_key = config.get(profile, 'aws_secret_access_key')
+            debug('Setting "aws_secret_access_key" from file {} as "secret_key"'.format(aws_credential_file))
+            Config().update_option('secret_key', config_unicodise(profile_secret_key))
+            
+            try:
+                # trying to read aws_session_token from credentials file
+                profile_access_token = config.get(profile, 'aws_session_token')
+                debug('Setting "aws_session_token" from file {} as "access_token"'.format(aws_credential_file))
+                Config().update_option('access_token', config_unicodise(profile_access_token))
+            except NoOptionError:
+                pass # do nothing, because "access_token" is optional 
         except IOError as e:
-            debug("Error %d accessing credentials file %s" % (e.errno,os.environ['AWS_CREDENTIAL_FILE']))
-        r_data = re.compile("^\s*(?P<orig_key>\w+)\s*=\s*(?P<value>.*)")
-        r_quotes = re.compile("^\"(.*)\"\s*$")
-        if len(cred_content)>0:
-            for line in cred_content.splitlines():
-                is_data = r_data.match(line)
-                if is_data:
-                    data = is_data.groupdict()
-                    if r_quotes.match(data["value"]):
-                        data["value"] = data["value"][1:-1]
-                    if data["orig_key"] == "AWSAccessKeyId" \
-                       or data["orig_key"] == "aws_access_key_id":
-                        data["key"] = "access_key"
-                    elif data["orig_key"]=="AWSSecretKey" \
```

Comment:
I have found a reference about that:
https://blog.csanchez.org/2011/05/
Looks like it used to be the case up to 2011 at least.
Its a little bit annoying to have to take care of that, but that would probably be better to still handle that case.
Note: a warning might still be issued.
But users could have inherited config files or found old tutorial on internet and still be using that.
It is bad to break user existing setups just for updates when it is not strictly necessary.